8.x/theme-development/helpers.md

```diff
@@ -2,4 +2,2097 @@
+        } catch (EmptyClipBoardException|InvalidTypeException $e) {
+            return false;
+        }
+    }
+}
+```
+
+
+### get_product_color_models()
+This function takes the product from the input and returns its colors.
+
+```php
+if (!function_exists('get_product_color_models')) {
+    function get_product_color_models(Product $product)
+    {
+        return Product::models($product, false)
+            ->orderBy('id', 'DESC')->groupBy('color_code')->get();
+    }
+}
+```
+
+
+### get_product_last_color()
+Sorts and returns the last color for the product.
+
+```php
+if (!function_exists('get_product_last_color')) {
+    function get_product_last_color(Product $product): Color
+    {
+        return $product->colors()->orderBy('id', 'DESC')->first();
+    }
+}
+```
+
+
+### get_product_accessories()
+This function takes the product and returns the accessories of that product.
+
+```php
+if (!function_exists('get_product_accessories')) {
+    function get_product_accessories(Product $product)
+    {
+        return $product->accessories()->mainModels()->visible()->get();
+    }
+}
+```
+
+
+### get_product_related_articles()
+Returns articles related to a product to the number taken from the input.
+
+```php
+if (!function_exists('get_product_related_articles')) {
+    function get_product_related_articles(Product $product, $type, int $count = 3)
+    {
+        $tags = $product->tags()->get()->pluck("id");
+        return Article::from($type)->whereHas("tags", function ($query) use ($tags) {
+            $query->whereIn("id", $tags);
+        })->latest()->take($count)->get();
+    }
+}
+```
+
+
+### get_product_related_products()
+Returns products related to a product to the specified number.
+
+```php
+if (!function_exists('get_product_related_products')) {
+    /**
+     * @param Product $product
+     * @param int $count
+     * @return mixed
+     */
+    function get_product_related_products(Product $product, int $count = 5)
+    {
+        $directory = $product->directory;
+        $leafProducts = new Collection();
+        $products = $directory->products()->mainModels()->visible()
+            ->where("is_active", true)->latest()->except($product->id)->take($count)->get();
+        while ($products->count() < $count and $directory->directory_id != null
+            and $leafProducts->count() < $count) {
+            $directory = $directory->parentDirectory;
+            $leafProducts = $directory->leafProducts()->mainModels()->visible()
+                ->where("is_active", true)->latest()->except($product->id)->take($count)->get();
+        }
+        return $products->merge($leafProducts)->unique()->take($count);
+    }
+}
+```
+
+
+### get_product_similar_products()
+The get_product_similar_products function declares a similarity key for a product. For example, a car with the bmw brand is considered. This function returns products that are similar to this machine but have a different brand.
```

Comment:
This function returns products with the exact specifications as this car but with a different brand.